### PR TITLE
UTY-764: World Commands Rewrite

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Added World Commands and all the associated types to match the current API.
Lots and lots of ~generated~ handwritten code.

Note that I've put the handling of the response ops into the `SpatialOSReceiveSystem` directly. If people prefer - I can refactor that out into a `WorldCommandsReceiveSystem` and make the dispatcher on the `SpatialOSReceiveSystem` internal such that the other system can register with it. 

#### Tests
It compiled! Tests coming soon ™️ 
#### Documentation
Also going to come soon
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.
@zeroZshadow

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs [hunk 1]
```diff
@@ -154,6 +170,158 @@ private void OnCommandResponse(CommandResponseOp op)
             specificDispatcher.OnCommandResponse(op);
         }
 
+        private void OnCreateEntityResponse(CreateEntityResponseOp op)
+        {
+            if (!createEntityStorage.CommandRequestsInFlight.TryGetValue(op.RequestId.Id, out var requestBundle))
+            {
+                throw new InvalidOperationException($"Could not find corresponding request for RequestId {op.RequestId.Id} and command CreateEntity.");
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs]: Possibly should make this an error for the logger, as there is nowhere for this to be caught and it doesn't put us in an unrecoverable state.

File: workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs [hunk 2]
```diff
@@ -154,6 +170,158 @@ private void OnCommandResponse(CommandResponseOp op)
             specificDispatcher.OnCommandResponse(op);
         }
 
+        private void OnCreateEntityResponse(CreateEntityResponseOp op)
+        {
+            if (!createEntityStorage.CommandRequestsInFlight.TryGetValue(op.RequestId.Id, out var requestBundle))
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs]: I think this is a good example of when `out var` is bad. I can't see what type this is and it's hard when reading code further down to see where this is created.

File: workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs [hunk 3]
```diff
@@ -13,24 +14,39 @@ public class SpatialOSReceiveSystem : ComponentSystem
...
 
         private readonly Dictionary<uint, ComponentDispatcherHandler> componentSpecificDispatchers =
             new Dictionary<uint, ComponentDispatcherHandler>();
 
-        private bool inCriticalSection = false;
+        private bool inCriticalSection;
 
+        private const string LoggerName = nameof(SpatialOSReceiveSystem);
         private const string UnknownComponentIdError = "Received an op with an unknown ComponentId";
+        private const string EntityNotFound = "No entity found for entity specified in op.";
```
Comment (samiwh) [inline workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs]: for `SpatialOS entity ID` maybe

File: workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs [hunk 4]
```diff
@@ -154,6 +170,158 @@ private void OnCommandResponse(CommandResponseOp op)
...
+            if (EntityManager.HasComponent<WorldCommands.CreateEntity.CommandResponses>(entity))
+            {
+                responses = EntityManager.GetComponentData<WorldCommands.CreateEntity.CommandResponses>(entity)
+                    .Responses;
+            }
+            else
+            {
+                var data = new WorldCommands.CreateEntity.CommandResponses
+                {
+                    Handle = WorldCommands.CreateEntity.ResponsesProvider.Allocate(World)
```
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs]: Yeah - that's a change in semantics however. Its related to efficient component tags/change filters that work. Etc.

[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs]: But yes we can pool the lists

File: workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs [hunk 5]
```diff
@@ -154,6 +170,158 @@ private void OnCommandResponse(CommandResponseOp op)
...
+            {
+                throw new InvalidOperationException($"Could not find corresponding request for RequestId {op.RequestId.Id} and command CreateEntity.");
+            }
+
+            var entity = requestBundle.Entity;
+            createEntityStorage.CommandRequestsInFlight.Remove(op.RequestId.Id);
+
+            if (!EntityManager.Exists(entity))
+            {
+                logDispatcher.HandleLog(LogType.Error, new LogEvent(EntityNotFound)
```
[the remark this answers is not in the document]
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Systems/SpatialOSReceiveSystem.cs]: Agreed - fixed

**Pull request merged**